### PR TITLE
Removed Ethers and making signers optional until need to sign

### PR DESCRIPTION
**Pull request by ekenigs**
### Description

Removed Ethers and making signers optional until need to sign

### Checklist

- [ ] If this requires a documentation change, I have created a PR in [moonbeam-docs](https://github.com/moonbeam-foundation/moonbeam-docs) repository.
- [ ] If this requires it, I have updated the Readme
- [ ] If necessary, I have updated the examples
- [ ] I have verified if I need to create/update unit tests
- [ ] I have verified if I need to create/update acceptance tests
- [ ] If necessary, I have run acceptance tests on this branch in CI

### CODE REVIEW

**Review by Rihyx — APPROVED** [review]
Looks good, but wrote a few small comments

File: packages/types/src/chain/parachain/Parachain.interfaces.ts [hunk 1]
```diff
@@ -16,4 +16,5 @@ export interface ChainAssetsData {
   minId?: ChainAssetId;
   palletInstance?: number;
   min?: number;
+  addrss?: string;
```
Comment (mmaurello) [inline packages/types/src/chain/parachain/Parachain.interfaces.ts:19]: typo / redundant? 
`address` is already in this interface

File: packages/sdk/src/sdk.interfaces.ts [hunk 2]
```diff
@@ -23,7 +18,10 @@ export interface TransferData {
   min: AssetAmount;
   source: SourceChainTransferData;
   swap(): Promise<TransferData | undefined>;
-  transfer(amount: bigint | number | string): Promise<string>;
+  transfer(
+    amount: bigint | number | string,
+    signers?: Signers,
```
[the remark this answers is not in the document]
Comment (ekenigs) [reply packages/sdk/src/sdk.interfaces.ts:23]: You can pass signer to `Sdk()` itself, or we can remove this option so we always receive on transfer.

File: packages/sdk/src/getTransferData/getSourceData.ts [hunk 3]
```diff
@@ -337,24 +324,20 @@ export async function getFee({
...
 
 export async function getContractFee({
+  address,
   balance,
   config,
   decimals,
-  evmSigner,
   chain,
 }: {
+  address: string;
```
[the remark this answers is not in the document]
Comment (ekenigs) [reply packages/sdk/src/getTransferData/getSourceData.ts:333]: Then I need to change them in the whole SDK, we can do it, but then I think we can do it in another PR.

File: packages/sdk/src/contract/contracts/Xtokens/Xtokens.ts [hunk 4]
```diff
@@ -1,92 +1,65 @@
...
-
-type XtokensContract = GetContractReturnType<typeof abi, PublicClient>;
+import { XTOKENS_ABI } from './XtokensABI';
 
 export class Xtokens implements TransferContractInterface {
   readonly defaultMoonChainAddress =
     '0x0000000000000000000000000000000000000804';
 
-  readonly address: Address;
+  readonly address: string;
```
[the remark this answers is not in the document]
Comment (ekenigs) [reply packages/sdk/src/contract/contracts/Xtokens/Xtokens.ts:12]: Answered above. Then we need to change it everywhere.

File: examples/sdk-simple/index.ts [hunk 5]
```diff
@@ -3,17 +3,19 @@ import { dot, moonbeam, polkadot } from '@moonbeam-network/xcm-config';
...
+import { Address, createWalletClient, http } from 'viem';
+import { privateKeyToAccount } from 'viem/accounts';
 
 // Moonbeam Signer ===========================================================
 
 const moonbeamPrivateKey = '';
-const provider = new ethers.WebSocketProvider(moonbeam.ws, {
-  chainId: moonbeam.id,
-  name: moonbeam.name,
+const account = privateKeyToAccount(moonbeamPrivateKey as Address);
```
[the remark this answers is not in the document]
Comment (ekenigs) [reply examples/sdk-simple/index.ts:13]: It is our example script on how to use the SDK. If you want to run it, you will need to provide keys.

**Pull request merged**